src/rules/utils/misc.ts

```diff
@@ -229,3 +229,63 @@ export const getFirstMatcherArg = (
 
   return followTypeAssertionChain(firstArg);
 };
+
+/* istanbul ignore next */
```

Comment:
why do we need to ignore coverage?

Reply:
because only one branch in each function should ever get used depending on what ESLint version you're using, and coverage is only collected against a single ESLint version at a time (technically we could merge reports for CI but that'd still be a pain locally)